lib/prawn/text/formatted/parser.rb

```diff
@@ -238,6 +236,14 @@ def self.array_from_tokens(tokens)
           end
           array
         end
+
+        def self.escape(text)
+          text.gsub('&', '&amp;').gsub('>', '&gt;').gsub('<', '&lt;')
```

Comment:
Since, we are refactoring this, maybe we can use a replacement has and spare use a few extra passes over the string?